libs/vscode/nx-project-view/src/lib/views/nx-project-tree-view.ts

```diff
@@ -0,0 +1,171 @@
+import { ProjectConfiguration } from '@nrwl/devkit';
+import { CliTaskProvider } from '@nx-console/vscode/tasks';
+import { getOutputChannel } from '@nx-console/vscode/utils';
+import { TreeItemCollapsibleState } from 'vscode';
+import {
+  NxFolderTreeItem,
+  NxProjectTreeItem,
+  NxTreeViewItem,
+} from '../nx-project-tree-item';
+import { BaseView, ProjectViewStrategy } from './nx-project-base-view';
+import { isDefined, PathHelper } from './nx-project-util';
+import path = require('node:path');
+
+export type TreeViewStrategy = ProjectViewStrategy<NxTreeViewItem>;
+type TreeViewMap = Map<string, [string, ProjectConfiguration][]>;
+
+export function createTreeViewStrategy(
+  cliTaskProvider: CliTaskProvider
+): TreeViewStrategy {
+  const listView = new TreeView(cliTaskProvider);
+  return {
+    getChildren: listView.getChildren.bind(listView),
+    getParent: listView.getParent.bind(listView),
+  };
+}
+
+class TreeView extends BaseView {
+  constructor(cliTaskProvider: CliTaskProvider) {
+    super(cliTaskProvider);
+  }
+
+  async getParent(element: NxTreeViewItem) {
```

Comment:
From what I gather, the `getParent` method is not really used to populate the TreeView but only for the `reveal` API: https://code.visualstudio.com/api/references/vscode-api#TreeDataProvider

One of two things would be great:
- Remove the `getParent` code to make it cleaner (lmk if I missed something and there's a good reason for it to be there)
- Add a 'Focus current project in projects view' or similar command that opens the project view to the project corresponding with the open file

Reply:
Removed